compass/landice/tests/greenland/mesh.py

```diff
@@ -203,18 +139,22 @@ def run(self):
                                 'westCentralGreenland'],
                           all_tags=False)
 
+        # Do some final validation of the mesh
+        ds = xr.open_dataset(self.mesh_filename)
         # Ensure basalHeatFlux is positive
-        data.variables['basalHeatFlux'][:] = np.abs(
-            data.variables['basalHeatFlux'][:])
+        ds["basalHeatFlux"] = np.abs(ds.basalHeatFlux)
         # Ensure reasonable dHdt values
-        dHdt = data.variables["observedThicknessTendency"][:]
-        dHdtErr = data.variables["observedThicknessTendencyUncertainty"][:]
+        dHdt = ds["observedThicknessTendency"]
         # Arbitrary 5% uncertainty; improve this later
         dHdtErr = np.abs(dHdt) * 0.05
-        # large uncertainty where data is missing
-        dHdtErr[np.abs(dHdt) > 1.0] = 1.0
-        dHdt[np.abs(dHdt) > 1.0] = 0.0  # Remove ridiculous values
-        data.variables["observedThicknessTendency"][:] = dHdt
-        data.variables["observedThicknessTendencyUncertainty"][:] = dHdtErr
-
-        data.close()
+        # Use threshold of |dHdt| > 1.0 to determine invalid data
+        mask = np.abs(dHdt) > 1.0
+        # Assign 100% uncertainty where data is missing
```

Comment:
```suggestion
        # Assign very large uncertainty where data is missing
```

This is absolute uncertainty, since we can't assign a percent uncertainty to missing data.